pacsman/pynetdicom_client.py

```diff
@@ -0,0 +1,399 @@
+            id_responses = _call_c_find_patients(assoc, 'PatientID',
+                                                 f'*{search_query}*')
+            # perform second search on patient name
+            name_responses = _call_c_find_patients(assoc, 'PatientName',
+                                                   f'*{search_query}*')
+
+            uid_to_result = {}
+            for (status, result) in chain(id_responses, name_responses):
+                logger.debug(status)
+                logger.debug(result)
+                if status.Status not in status_success_or_pending:
+                    raise Exception('Patient C-FIND Failure Response: 0x{0:04x}'.format(status.Status))
+                if result:
+                    # remove non-unique Study UIDs
+                    #  (some dupes are returned, especially for ID search)
+                    uid_to_result[result.StudyInstanceUID] = result
+
+            # separate by patient ID, count studies and get most recent
+            patient_id_to_info = {}
+            for study in uid_to_result.values():
+                patient_id = study.PatientID
+                study_id = study.StudyInstanceUID
+                if id in patient_id_to_info:
+                    if study.StudyDate > patient_id_to_info[patient_id].most_recent_study:
+                        most_recent_study = study.StudyDate
+                    else:
+                        most_recent_study = patient_id_to_info[patient_id].most_recent_study
+
+                    prev_study_ids = patient_id_to_info[patient_id].num_studies
+
+                    info = PatientInfo(first_name=study.PatientName.given_name,
+                                       last_name=study.PatientName.family_name,
+                                       dob=study.PatientBirthDate,
+                                       patient_id=patient_id,
+                                       most_recent_study=most_recent_study,
+                                       study_ids=prev_study_ids + [study_id])
+                else:
+                    info = PatientInfo(first_name=study.PatientName.given_name,
+                                       last_name=study.PatientName.family_name,
+                                       dob=study.PatientBirthDate,
+                                       patient_id=patient_id,
+                                       most_recent_study=study.StudyDate,
+                                       study_ids=[study_id])
+                patient_id_to_info[patient_id] = info
+
+            return list(patient_id_to_info.values())
+
+    def studies_for_patient(self, patient_id):
+        ae = AE(ae_title=self.client_ae, scu_sop_class=QueryRetrieveSOPClassList)
+
+        with association(ae, self.pacs_url, self.pacs_port) as assoc:
+            responses = _call_c_find_patients(assoc, 'PatientID', patient_id)
+
+            study_ids = []
+            for (status, result) in responses:
```

Comment:
It seems like these lines:

```
+            for (status, result) in responses:
 +                if status.Status not in status_success_or_pending:
 +                    raise Exception('Studies C-FIND Failure Response: 0x{0:04x}'.format(status.Status))
```

are repeated a few times. Perhaps we should create a helper function for this?  Also, what happens if a result is pending?